src/clustering.hpp

```diff
@@ -37,6 +37,16 @@ namespace hmat {
 class ClusterTree;
 class DofCoordinates;
 class ClusteringAlgorithm;
+class AxisAlignedBoundingBox;
+
+class AxisAlignClusteringAlgorithm {
+protected:
+  void sortByDimension(ClusterTree& node, int dim) const;
+  virtual AxisAlignedBoundingBox* getAxisAlignedBoundingbox(const ClusterTree& node) const;
+  int largestDimension(const ClusterTree& node) const;
+  double volume(const ClusterTree& node) const;
+  void sort(ClusterTree& current, int axisIndex, int spatialDimension) const;
+};
 
```

Comment:
I would say that AxisAlignClusteringAlgorithm should inherit from ClusteringAlgorithm. This would avoid multiple inheritance.